### PR TITLE
RetroAchievements - Rich Presence Tweaks

### PR DESCRIPTION
**Pull request by LillyJadeKatrin**
A handful of small tweaks/fixes pertaining to Rich Presence in the Achievement Manager.

### CODE REVIEW

**Comment by iwubcode** [comment]
As a general note, I'd suggest we move to `std::steady_clock` over `std::time`.

For instance, instead of:

```
if (difftime(current_time, m_last_ping_time) > 120)
```

you'd have:

```
if (current_time - m_last_ping_time > std::chrono::minutes{2})
```

I didn't look to see if this is the only place that uses timestamps in the RetroAchivement code, so it might be more appropriate for a separate PR but thought I'd mention it.

File: Source/Core/Core/AchievementManager.cpp [hunk 1]
```diff
@@ -1341,17 +1335,44 @@ void AchievementManager::ActivateDeactivateAchievement(AchievementId id, bool en
...
-        this, nullptr);
-  });
+  if (!m_is_game_loaded)
+    return;
+  auto current_time = std::chrono::steady_clock::now();
+  if (current_time - m_last_rp_time > std::chrono::seconds{10})
+  {
+    if (Config::Get(Config::RA_RICH_PRESENCE_ENABLED))
+    {
+      Core::RunAsCPUThread([&] {
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:1347]: #12658 is removing this function.  Maybe switch to `CPUThreadGuard`?

File: Source/Core/Core/AchievementManager.cpp [hunk 2]
```diff
@@ -1341,17 +1335,44 @@ void AchievementManager::ActivateDeactivateAchievement(AchievementId id, bool en
...
+            [](unsigned address, unsigned num_bytes, void* ud) {
+              return static_cast<AchievementManager*>(ud)->MemoryPeeker(address, num_bytes, ud);
+            },
+            this, nullptr);
+      });
+    }
+    m_last_rp_time = current_time;
+    m_update_callback();
+
+    if (current_time - m_last_ping_time > std::chrono::minutes{2})
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:1360]: Not entirely sure what rich presence is but is there a reason the get and the ping are in the same block?

File: Source/Core/Core/AchievementManager.cpp (hunk 1 above)
Comment (mitaclaw) [reply Source/Core/Core/AchievementManager.cpp:1347]: The reason for its removal is that a braced scope containing a `CPUThreadGuard` is just as capable as the scope of the function passed to `RunAsCPUThread` (and easier to reason about, imo).

**Comment by LillyJadeKatrin** [comment]
Closing due to deprioritizing. Will re-examine once #12639 is complete and merged.

**Pull request closed without merging**